app/controllers/checkouts_controller.rb

```diff
@@ -3,13 +3,26 @@ class CheckoutsController < ApplicationController
 
   def create
     @book = Book.find(params[:book_id])
-    checkout = Checkout.create(book: @book, user: current_user)
+    if CheckOutBook.process(book: @book, user: current_user)
+      flash[:success] = t('.successful_message')
```

Comment:
@derekprior, Is it a better idea to have the Ruby service object handle success and error/failure? The controller was so sleek until I added the `if else` clause.